velox/vector/VectorSaver.h

```diff
@@ -66,4 +66,12 @@ std::string restoreStringFromFile(const char* FOLLY_NONNULL filePath);
 std::optional<std::string> generateFilePath(
     const char* FOLLY_NONNULL basePath,
     const char* FOLLY_NONNULL prefix);
+
+/// Generates a file directory specified by 'dirPath'. The generation will be
+/// recursive. Non-exist parent directories will also be created. Returns true
+/// if creation is successful, false otherwise. Error message will be printed if
+/// creation is unsuccessful, but already created directories will not be
+/// removed.
+bool generateFileDirectory(const std::string& dirPath);
```

Comment:
just looked up the cpp filesystem interface, turns out you can use std::filesystem::create_directories() that recursively generates directories.

Reply:
Turns out so! Thanks for the pointer. Let me make the changes

Reply:
FYI, see velox/common/base/Fs.h